manageiq-operator/api/v1alpha1/helpers/miq-components/kafka.go

```diff
@@ -282,7 +365,9 @@ func KafkaDeployment(cr *miqv1alpha1.ManageIQ, scheme *runtime.Scheme) (*appsv1.
 					Labels: deploymentLabels,
 					Name:   "kafka",
 				},
-				Spec: corev1.PodSpec{},
+				Spec: corev1.PodSpec{
+					Hostname: "kafka",
+				},
```

Comment:
By default the hostname FQDN is the pod name however, it needs to match the CN used in the certs. See https://github.com/bitnami/containers/blob/main/bitnami/kafka/README.md#security